pkg/cmd/server/origin/master_config.go

```diff
@@ -175,16 +177,18 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
 	kubeletClientConfig := configapi.GetKubeletClientConfig(options)
 
 	// in-order list of plug-ins that should intercept admission decisions (origin only intercepts)
-	admissionControlPluginNames := []string{"ProjectRequestLimit", "OriginNamespaceLifecycle", "PodNodeConstraints", "BuildByStrategy", "OriginResourceQuota"}
+	admissionControlPluginNames := []string{"ProjectRequestLimit", "OriginNamespaceLifecycle", "PodNodeConstraints", "BuildByStrategy", imageadmission.PluginName, "OriginResourceQuota"}
 	if len(options.AdmissionConfig.PluginOrderOverride) > 0 {
 		admissionControlPluginNames = options.AdmissionConfig.PluginOrderOverride
 	}
 
+	registry := quota.NewRegistryForAdmission(privilegedLoopbackOpenShiftClient, options.ImagePolicyConfig.MaxImagesBulkImportedPerRepository)
```

Comment:
`NewOriginQuotaRegistry` would do.

On Wed, Mar 30, 2016 at 4:39 AM, Michal Minar <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/master_config.go
> <https://github.com/openshift/origin/pull/8195#discussion_r57853858>:
>
> >  	if len(options.AdmissionConfig.PluginOrderOverride) > 0 {
> >  		admissionControlPluginNames = options.AdmissionConfig.PluginOrderOverride
> >  	}
> >
> > +	registry := quota.NewRegistryForAdmission(privilegedLoopbackOpenShiftClient, options.ImagePolicyConfig.MaxImagesBulkImportedPerRepository)
>
> Doesn't this mean that I'm setting a partially complete QuotaRegistry?
>
> It's a complete registry for OpenShift resources.
>
> I don't see how this handles the kube types.
>
> It doesn't. There's a second quota admission plugin
> <https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/plugin/pkg/admission/resourcequota/admission.go#L44>
> running for k8s resources.
>
> If it's too hard to build the whole thing, name this such that its obvious
> it only works for a subset of types.
>
> Hmm, would NewOriginQuotaRegistryForAdmission() do?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/009f353ce0f5825f8e3821be026ae9f687ab66dc#r57853858>
>